### ISSUE TITLE
Gimlet manifest generation - CUE support

### ISSUE THREAD

**Issue by laszlocph** [gimlet-io/gimlet-cli#41]
- Today Gimlet works with Gimlet manifest files stored in the `.gimlet/` folder.
- Gimlet manifests are yaml files.
- Today we follow the one manifest per yaml file approach.
- Yaml has yaml anchors to promote some DRYness, however that doesn't work across yaml files

All the above means that if an app is deployed in several configurations, it results in large scale code duplication.
- Today there are Gimlet users, generating 50-100 manifests from a Python script
- Also another user reported a monorepo approach that has ~10 yamls in the `.gimlet/` folder. All yamls being largely the same

The goal of this task is to reduce code duplication and introduce reusable configuration pieces.

The goal is also not to introduce an other level abstraction, but solve this with a configuration language option.

**Comment by laszlocph** [comment]
One option is using CUE for this.

There would be several `.cue` files in the `.gimlet/` folder and each can generate an array of gimlet manifest yamls.

It can be any cue file, that is generating a 

```yaml
- app: xxx
  env: ...
- app yy
  env: xx
```
yaml that is a valid Gimlet manifest.

Such cue file may look like this:

```cue
package gimletio

_image:     "mycompany/myapp
_tag:       "v3.5.1"
_vars: {
	SOME_VARS: "that are the same for all apps"
        SOME_MORE_VAR: "That are the same"
        ...
}

_instances: [
	{
		app: "first"
		values:
			replicas: 3
                        vars:
                             THIS_ONE: "has one more var, cue will figure it out"
	},
	{
		app: "second"
		values:
			replicas: 3
	},
]

// The rendered output from instances+template
// list comprehension syntax
[ for instance in _instances {instance & #template & #gimletManifest}]

// template from `cue import -f -o gimlet.cue -p "gimletio" gimlet.yaml`
// fields that are variable, we add a type, and add them default value, or make the optional, etc
// maybe zap it together with the official Gimlet cue files? no def within values
#template: {
	env:       "production"
	namespace: "production"
	chart:
		repository: https://chart.onechart.dev
                name: onechart
                version: 0.32.0
	values: {
		replicas: int
		image: {
			repository: _image
			tag:        _tag
			pullPolicy: "Always"
		}
		resources: {
			requests: {
				cpu:    "100m"
				memory: "1250Mi"
			}
			limits: {
				cpu:    "1000m"
				memory: "2000Mi"
			}
		}
		vars:          _vars
		...
	}
	...
}


#gimletManifest: {
	app:       string
	env:       string
	namespace: string
	chart:
		name: string
	values: {
		...
	}
	...
}
```

The key is the list comprehension: `[ for instance in _instances {instance & #template & #gimletManifest}]` which renders all instances, and zaps together the partial instance data, with the app specific `#template` schema, and with the official #gimletManifest schema.

`gimlet manifest template` would understand cue files, and will act along the lines of `cue export gimlet.cue  --out yaml`, expecting a yaml array as an output.

It should also support Gimlet variables, so the `_tag:       "v3.5.1"` part from the above snippet would be `_tag:       "{{ .GITHUB_SHA }}"` and gimlet would resolve these before processing cue.

**Comment by laszlocph** [comment]
Will be released in Gimlet CLI v0.12.0 and GimletD v0.8.0
